packages/sources/universal-v3/src/endpoint/lambda.ts

```diff
@@ -0,0 +1,203 @@
+  },
+  secrets: {
+    description: 'Base64 bytestring representing an encrypted secrets object in base64 format',
+    dependsOn: ['secretsOwner', 'subscriptionId'],
+  },
+  secretsOwner: {
+    description: 'Owner of the encrypted secrets used for signature verification',
+    dependsOn: ['secrets', 'subscriptionId'],
+  },
+  args: {
+    description: 'Array of on-chain arguments which are passed to the source code',
+    type: 'array',
+  },
+  codeLocation: {
+    description:
+      'Location of user-provided code, encoded as an integer represeting a enum (0 = on-chain)',
+    type: 'number',
+    options: [0],
+  },
+  secretsLocation: {
+    description:
+      'Location of user-provided secrets, encoded as an integer represeting a enum (0 = on-chain)',
+    type: 'number',
+    options: [0],
+  },
+  language: {
+    description:
+      'Language of the user-provided code, encoded as an integer represeting a enum (0 = javascript)',
+    type: 'number',
+    options: [0],
+  },
+}
+
+interface Input {
+  source: string
+  subscriptionId: string
+  requestId: string
+  numAllowedQueries?: number
+  maxResponseBytes?: number
+  args?: string[]
+  secrets?: base64bytes
+  secretsOwner?: string
+  codeLocation?: Location.Onchain
+  secretsLocation?: Location.Onchain
+  language?: Language.JavaScript
+}
+
+type base64bytes = string
+
+enum Location {
+  Onchain = 0,
+}
+
+enum Language {
+  JavaScript = 0,
+}
+
+interface Request extends RequestGenerics {
+  Params: Input
+}
+
+interface AdapterIO extends TransportGenerics {
+  Request: Request
+  Response: Response
+  CustomSettings: typeof customSettings
+}
+
+interface LambdaTransport extends Transport<AdapterIO> {
+  lambdaController: LambdaController
+}
+
+class Lambda implements LambdaTransport {
+  lambdaController!: LambdaController
+
+  initialize = async (
+    _dependencies: TransportDependencies<AdapterIO>,
+    config: AdapterConfig<typeof customSettings>,
+  ): Promise<void> => {
+    this.lambdaController = new LambdaController(config)
+    await this.lambdaController.initalize()
+    return
+  }
+
+  // This function handles each request sent to the EA
+  foregroundExecute = async (
+    req: AdapterRequest<Request>,
+    config: AdapterConfig<typeof customSettings>,
+  ): Promise<AdapterResponse<Response>> => {
+    const requestId = req.requestContext.data.requestId
+    const source = req.requestContext.data.source
+    const subscriptionId = req.requestContext.data.subscriptionId
+    const secrets = req.requestContext.data.secrets
+    const secretsOwner = req.requestContext.data.secretsOwner
+    const args = req.requestContext.data.args
+    const numAllowedQueries = req.requestContext.data.numAllowedQueries
+    const maxResponseBytes = req.requestContext.data.maxResponseBytes
+
+    const responseBuilder = new ResponseBuilder(config)
+
+    let decryptedSecrets: Record<string, unknown> = {}
```

Comment:
Secrets have to be strings right?
```suggestion
    let decryptedSecrets: Record<string, string> = {}
```